v-next/hardhat-build-system/src/internal/builtin-tasks/utils/solidity-files-cache.ts

```diff
@@ -0,0 +1,135 @@
+import type { SolcConfig } from "../../types/index.js";
+import debug from "debug";
+import fsExtra from "fs-extra";
+import * as t from "io-ts";
+import { deepEqual } from "fast-equals";
```

Comment:
To pull this in directly from utils would mean making the chain of function calls above the cache `async`, is that what we intend with an async deepEqual?

Reply:
This is probably a question for @alcuadrado, but I believe we should be fine. The asynchronicity comes from dynamically importing the `fast-equals` package inside `deepEqual`. However, after the initial load, it is cached, so subsequent loads should be fast. The only potential disadvantage is the extra boilerplate required by all the `async`/`await` statements.

Reply:
We discussed this and decided to defer the decision